do not merge: WIP commits for debugging a production issue

bdarnell opened the pull request:
I suspect that contention on this semaphore is causing failed
heartbeats in a production cluster.

This PR is based on release-1.0 so it can be tried immediately; if the experiment is successful I'll polish it up before merging and make a version for `master`.

bdarnell committed:
storage: Use separate locks when updating our own liveness

I suspect that contention on this semaphore is causing failed
heartbeats in a production cluster.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16564)
<!-- Reviewable:end -->

bdarnell committed:
WIP disable MergeAndDestroy assertion

bdarnell committed:
WIP increase sendnexttimeout

bdarnell committed:
WIP add a limit on how many keys can be GC'd at once

Note that because all of these keys are behind the GCThreshold, it
doesn't matter that we're not deleting them in FIFO order.

bdarnell committed:
WIP limit batch size in intent resolver

bdarnell commented:
The important commits in this branch have been broken out into separate PRs: the first commit is #16918 (master-only), the fourth is #16637 (master) and #16735 (release-1.0), and the fifth is #16632 (master) and #16739 (release-1.0).
